### PR TITLE
Fix/#9120 VDatePicker range performance

### PR DESCRIPTION
**Pull request by YipingRuan**
## Description
Check date selection for range in the current month only.
Currently smooth to support range as big as '0000-01-01', '2019-09-20', should be enough.
## Motivation and Context
https://github.com/vuetifyjs/vuetify/issues/9120

## How Has This Been Tested?
visually

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-date-picker v-model="dates" range width="290" class="mt-4"></v-date-picker>
    v-model value: {{ dates }}
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      dates: ['0000-01-01', '2019-09-20'],
    })
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by nekosaur** [comment]
1. What is this code even supposed to do?
2. Why cache manually instead of using a computed prop?

**Comment by YipingRuan** [comment]
> 1. What is this code even supposed to do?
> 2. Why cache manually instead of using a computed prop?

Thanks. Moved the logic to date-picker-table.ts for the displaying month only

File: packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts [hunk 1]
```diff
@@ -100,7 +101,7 @@ export default mixins(
     },
     genButton (value: string, isFloating: boolean, mouseEventType: string, formatter: DatePickerFormatter) {
       const isAllowed = isDateAllowed(value, this.min, this.max, this.allowedDates)
-      const isSelected = value === this.value || (Array.isArray(this.value) && this.value.indexOf(value) !== -1)
+      const isSelected = value === this.value || (Array.isArray(this.value) && (this.value.indexOf(value) !== -1 || this.isInRange(value)))
```
Comment (nekosaur) [inline packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts]: refactor all of the logic into a `isSelected` function or something similar

File: packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts [hunk 2]
```diff
@@ -192,5 +193,13 @@ export default mixins(
         directives: [touchDirective],
       }, [transition])
     },
+    isInRange (value: string): boolean {
+      if (this.range && this.value.length === 2) {
+        const [from, to] = [...this.value as string[]].sort()
+        return from <= value && value <= to
```
Comment (nekosaur) [inline packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts]: This is all down to personal preferences, but for me
```
value >= from && value <= to
```
is much clearer

File: packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts [hunk 3]
```diff
@@ -192,5 +193,13 @@ export default mixins(
         directives: [touchDirective],
       }, [transition])
     },
+    isInRange (value: string): boolean {
+      if (this.range && this.value.length === 2) {
+        const [from, to] = [...this.value as string[]].sort()
```
Comment (nekosaur) [inline packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts]: Add `PropValidator<string | string[]>` to `value` prop and you should be able to remove `as string[]`

File: packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts (hunk 2 above)
Comment (KaelWD) [reply packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts]: I prefer this way, it's like in mathematics from &leq; value &leq; to

File: packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts (hunk 3 above)
Comment (YipingRuan) [reply packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts]: This is super convenient.

**Comment by TheInvoker** [comment]
when will this be released?

**Comment by johnleider** [comment]
Is there a reason this is to dev instead of master?

**Comment by YipingRuan** [comment]
> Is there a reason this is to dev instead of master?

No, bugs should -> master. 
Forgot this new feature is already there.

**Comment by TheInvoker** [comment]
So is this coming in a version after 2.1.2?

**Pull request merged**